[13.0][MIG] account_invoice_import

yvaucher opened the pull request:
With changes of model from `account.invoice` to `account.move`

TODO:

- [x] Fix dashboard
- [x] Check if account_id is still necessary on taxes (in unit tests)

Replaced by #727

alexis-via committed:
Rename module from account_invoice_pdf_import to account_invoice_import

FIX raise error when we don't identify a partner

Update README

alexis-via committed:
Update FR translation

alexis-via committed:
Add instructions to use my PPA for poppler-utils on Ubuntu 14.04 LTS + instructions OCR

alexis-via committed:
Add support for direct import of XML files (YAML test provided)

Rename test/pdf/ to test/invoices/ to host XML invoices too

Commit:
Sometimes, it's easier to extract the total tax amount instead of the total untaxed (depends on invoice layout)

alexis-via committed:
Add an explaination about date parsing in invoice2data

Commit:
Add optional support for start/end dates

alexis-via committed:
Adapt to latest changes in invoice2data

PEP8 stuff

alexis-via committed:
Only change name_get of invoice for our use-case, not everywhere

alexis-via committed:
Update README with latest changes in invoice2data lib

Commit:
Add support for partner_email to match partners

PEP8 fix

alexis-via committed:
Update installation procedure.

alexis-via committed:
Add modules base_zugferd, account_invoice_import_zugferd and account_invoice_zugferd (see README.rst for more info)

Move the ZUGFeRD/CII-specific code of account_invoice_import to account_invoice_import_zugferd

alexis-via committed:
Log raw result of invoice2data (before we make modifications)

Disable test of invoice2data, because this lib is difficult to install

alexis-via committed:
Fix error in README spotted by Roel Adriaans

alexis-via committed:
Add module account_invoice_import_ubl

Add support for UBL refunds

If you have an UBL refund that I could use to test, please contact me !

alexis-via committed:
Use base_vat_sanitized

Convert YAML test to unittest

alexis-via committed:
Add support for import of e-fff invoices (Belgian standard)

alexis-via committed:
Add support for local invoice template for invoice2data, see README

alexis-via committed:
Small usability improvement

alexis-via committed:
Checks that the directory really exists

alexis-via committed:
Extract regular PDF invoice import from account_invoice_import to a dedicated module account_invoice_import_invoice2data

Update README.rst and headers to latest OCA conventions.

alexis-via committed:
Adapt code to plug the module 'ovh_supplier_invoice'

Better key names in the parsed_inv dict
parsed_inv doesn't need to be JSON serializable anymore (small drawback: the invoice is parsed a second time on the second step... but the second step is rarely used)

alexis-via committed:
Fix product matching

Move code from account_invoice_import_invoice2data to account_invoice_import

alexis-via committed:
FIX PEP8 and account_invoice_import_invoice2data unittests

Update REAME and some interface strings about UBL being an ISO standard
Small code changes

alexis-via committed:
Add module base_business_document_import and start to move some code to it.

alexis-via committed:
Better/cleaner pivot invoice dict format (no more IDs in the pivot invoice dict, cleaner organisation)

Code refactoring: move code in base_business_document_import, factorise code for tax matching (it was duplicated in UBL and ZUGFeRD)
Now support PDF with embedded UBL XML file
Enable unittests on account_invoice_import_ubl
More absolute xpath in account_invoice_import_ubl instead of relative xpath

WARNING: these are big changes, I may have broken a few details

alexis-via committed:
Add module base_ubl (common methods to generate and parse UBL files)

Adapt module account_invoice_import_ubl to use the new base_ubl module
Small fixes

alexis-via committed:
Add module base_phone_business_document_import

Add unitests in base_business_document_import
Small code enhancements/simplifications

alexis-via committed:
Add a unittest on _match_uom

Code cleanup/minor changes

alexis-via committed:
Add module base_business_document_import_stock

Rename key 'quantity' to 'qty' in all parsing dicts['lines']
Add common methods compare_lines() and post_create_or_update()
Make sure price_unit is always untaxed in UBL XML files

alexis-via committed:
Update invoice lines on "Update Existing" invoice import

Add button "Update from File" on supplier invoice form
Use float_round() instead of round()

alexis-via committed:
Fix crash

alexis-via committed:
Add multi-company ir.rule on account.invoice.import.config

alexis-via committed:
FIX Don't use _inherit = ['business.document.import']

because we want to have access to the code of the modules that inherit business.document.import

alexis-via committed:
Fix method call

Commit:
[IMP] handle the case where the xml file is generated with mime type
'text/xml'

alexis-via committed:
Add support for extraction + matching on website

PEP8 fix

alexis-via committed:
FIX two stupid bugs

Commit:
8.0 Add support for partner bank matching on invoice update (#6)

Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

Commit:
8.0 Use try/except when importing external libs (#8)

[FIX] LINT

Use try/except when importing external libs
Remove self.ensure_one() that has nothing to do in an api.model method

hbrunn committed:
[MIG] Make modules uninstallable

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[ADD] migration of account_invoice_import_invoice2data and dependencies

alexis-via committed:
Add support for 'origin' field in invoice import

hbrunn committed:
[FIX] product.product#ean13 was renamed to barcode

hbrunn committed:
[FIX] tests

alexis-via committed:
Port base_business_document_import* to v10

Also port all the modules that generate the XML documents: account_invoice_ubl, account_invoice_zugferd, purchase_order_ubl and sale_order_ubl

Add active buttons on invoice import config

Note: it requires this fix on odoo 10: https://github.com/odoo/odoo/pull/15649

Port account_invoice_import_factur-x and account_invoice_import_ubl to v10.0

base_business_document_import: Add support for the creation of res.bank

Port purchase_order_import* to v10.0

Add ubl invoice generation option in accounting config page

Update to OCA coding standards

alexis-via committed:
Allow importing invoice without invoice_import_config on partner (usefull for third-party modules that use account_invoice_import)

Code cleanup and optimisations

alexis-via committed:
Several fixes in account_invoice_import

Reorganise menu entries for invoice import and update labels

alexis-via committed:
Improve XML for Factur-X

Make country_id field required on res.partner form view

alexis-via committed:
Add first support for the import of Factur-X invoices

Add support for start/end dates on lines
Better handling of invalid IBANs

alexis-via committed:
Add support for allowance and charges on Factur-X invoice import

To avoid headaches with invalid files, now check XSD of XML invoices before parsing it upon import
Move Factur-X/zugferd xsd to base_zugferd

alexis-via committed:
Fix bug #16 : add support for adjustment lines (per line and global)

Add support for Tax Due Date on invoice import

alexis-via committed:
account_invoice_factur-x: now generate refunds with document type = 381 instead of document type = 380 with negative amounts

Improve the code of refund import to support both negative invoices and positive refunds

alexis-via committed:
Pre-process parsed_inv even when calling directly _create_invoice()

alexis-via committed:
Add support for importing customer invoices (bug #24)

Add basic unit-tests in account_invoice_import

alexis-via committed:
Improve + modularize generation of UBL (up-port from v8 PR)

Add script mass_invoice_import.py
Code cleanup

alexis-via committed:
Finalise the use of the FacturX lib

One of the zugferd-specific code has been standardised in UNECE, so moved to the account_tax_unece module

alexis-via committed:
account_invoice_import: workaround a framework bug

Fix update invoice via import wizard

alexis-via committed:
account_invoice_import: Add partner_type to _match_partner method for better support out customer invoice import

Fix test suite of account_invoice_import (Agrolait is only a customer in v10)

alexis-via committed:
Partner <-> Invoice import config : switch M2O <-> O2M

alexis-via committed:
Add migration script

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0] Update to work with latest version of invoice2data (#51)

* Update to work with latest version of invoice2data

* Add requirements.txt file

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0] Several fixes + import invoices with VAT in a company that cannot deduct VAT (#64)

* Move code to a new dedicated method
* FIX update of existing draft invoice
  Improve error message of partner matching
  Move hook of partner matching to put it after VAT matching, which is the
  most reliable
* Add support for importing an invoice with VAT in a company that cannot deduc VAT
  Display nice error when account is missing on product
* Improve compliancy with Factur-X standard (using schematron)

oca-travis committed:
[UPD] Update account_invoice_import.pot

alexis-via committed:
Fix import of sale order with price_source='pricelist' when pricelist has visible discounts

Code improvements in sale_order_import
Add unit tests in sale_order_import
Use display_name instead of name_get()[0][1]

alexis-via committed:
Add support for multiple invoice import configs in invoice import wizard

Update other modules accordingly

alexis-via committed:
Add O2M field from import config to download config

Code update and cleanup

Commit:
[FIX] include some of the fixes proposed in #64

alexis-via committed:
Finalize the development account_invoice_download

Update the account_invoice_download_weboob following the changes in account_invoice_download
Add README.rst for the 2 modules

alexis-via committed:
Add support for additional module-specific weboob config parameters

Add weboob module management
Improve log handling (explicit error when wrong pwd)
account_invoice_import: by default, set tax to default tax of account when invoice_line_method = 1line_no_product

alexis-via committed:
account_invoice_import: add a mail gateway

Update text displayed in the invoice import wizard and make list of
supported formats modular (like in bank statement import)

alexis-via committed:
Add ability to run cron invoice download from admin for all companies in multi-company environnement

Fix my previous commit on sequence field on invoice import config (caused a crash when creating a new record)
Usability improvements on download config

alexis-via committed:
Add multi-company support in the invoice import mail gateway

alexis-via committed:
Check the VAT number of the destination partner, to make sure the business document is imported in the right company

oca-travis committed:
[UPD] Update account_invoice_import.pot

Commit:
[11.0][MIG] account_invoice_import

Commit:
[FIX] clean rebase  from V10 that where wrong ...

Commit:
[FIX] utf8 comment in python3

astirpe committed:
Fix base64 encode/decode + remove supplier_invoice_number + code review

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

OCA-git-bot committed:
[ADD] icon.png

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-11.0/edi-11.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-11-0/edi-11-0-account_invoice_import/

OCA-git-bot committed:
[UPD] README.rst

astirpe committed:
[12.0][MIG] account_invoice_import

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/

alexis-via committed:
account_invoice_download: start OCA import bill wizard when using the "Import Bill" button on journal dashboard

tv-openbig committed:
Added translation using Weblate (German)

tv-openbig committed:
Translated using Weblate (German)

Currently translated at 92.4% (97 of 105 strings)

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/de/

alexis-via committed:
[MIG] account_invoice_import_facturx to v12

Update technical name of module to remove '-'
Up-port PR #78 Auto-creation of bank accounts upon invoice import is now optional
Fix returned action when using the import wizard

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/

marylla committed:
Translated using Weblate (German)

Currently translated at 91.5% (97 of 106 strings)

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/de/

yvaucher committed:
[IMP] account_invoice_import: black, isort, prettier

yvaucher committed:
[MIG] account_invoice_import: Migration to 13.0

yvaucher commented:
@alexis-via FYI working on it

yvaucher committed:
[IMP] account_invoice_import: parsed invoice hook

Add a hook to inject parsed invoice from othe means than a file

yvaucher committed:
[IMP] account_invoice_import: Missing partner hook

Add a hook for an action on missing partner

yvaucher committed:
Add a quick import invoice button

A button is added on the Vendor Bill form to import invoices in serie
This button can be accessed through the keyboard shortcut with Alt + M (or an other letter depending on the modules installed on the server

yvaucher committed:
Restore dashboard button overwrite

CasVissers-360ERP commented:
@yvaucher Is this still WIP or can it be validated?

CasVissers-360ERP commented on account_invoice_import/security/rule.xml:
```diff
@@ -0,0 +1,14 @@
+        <field name="model_id" ref="model_account_invoice_import_config" />
+        <field
+            name="domain_force"
+        >['|', ('company_id', '=', False), ('company_id', 'child_of', [user.company_id.id])]</field>
```
Should be ['|',('company_id', 'in', company_ids),('company_id','=',False)]

yvaucher replied:
Thanks

CasVissers-360ERP reviewed (changes requested):
Record rule is created incorrectly.

yvaucher commented:
@CasVissers-360ERP This is somewhat ready, functional review would be good though, I'll open it.

NL66278 committed:
[FIX] account_invoice_import: fix headers

NL66278 committed:
[FIX] account_invoice_import: Reduce method complexity.

NL66278 committed:
[FIX] account_invoice_import. Do not shadow config module with variable.

etobella committed:
[IMP] base_business_document_import: edi integration

etobella committed:
[IMP] account_invoice_import: Integration with EDI

astirpe commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1190 @@
+        readonly=True,
+    )
+    amount_untaxed = fields.Float(
+        string="Total Untaxed", digits=dp.get_precision("Account"), readonly=True
```
```suggestion
        string="Total Untaxed", digits="Account", readonly=True
```

astirpe commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1190 @@
+        string="Total Untaxed", digits=dp.get_precision("Account"), readonly=True
+    )
+    amount_total = fields.Float(
+        string="Total", digits=dp.get_precision("Account"), readonly=True
```
```suggestion
        string="Total", digits="Account", readonly=True
```

astirpe commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1190 @@
+            "type": parsed_inv["type"],
+            "company_id": company.id,
+            "invoice_origin": parsed_inv.get("origin"),
+            "invoice_payment_ref": parsed_inv.get("invoice_number"),
```
```suggestion
            "ref": parsed_inv.get("invoice_number"),
```

CasVissers-360ERP replied:
Why do you suggest this change?

astirpe replied:
Is "invoice_payment_ref" the right field to be used here?

CasVissers-360ERP replied:
Yes I think using the supplier's invoice number as payment reference is the correct workflow.

astirpe replied:
OK then forget my suggested change :+1:

astirpe commented on account_invoice_import/security/rule.xml:
```diff
@@ -0,0 +1,12 @@
+    <record id="account_invoice_import_config_rule" model="ir.rule">
+        <field name="name">Import Supplier Invoices Configuration multi-company</field>
+        <field name="model_id" ref="model_account_invoice_import_config" />
+        <field name="domain_force">['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]</field>
```
As per https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0, this needs a migration script.

astirpe reviewed (changes requested):
This is an initial code review, I didn't check this PR in deep. What is the status of this PR? Still something left to be completed?

StefanRijnhart commented:
For the record, there is also https://github.com/OCA/edi/pull/273 and there is a discussion about integration with the `edi` module.

CasVissers-360ERP commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1190 @@
+        if float_compare(
+            invoice.amount_total, parsed_inv["amount_total"], precision_rounding=prec
+        ):
+            if not invoice.tax_line_ids:
```
AttributeError: 'account.move' object has no attribute 'tax_line_ids'

CasVissers-360ERP commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1190 @@
+        for entry in ["amount_untaxed", "amount_total"]:
+            parsed_inv[entry] = float_round(parsed_inv[entry], precision_digits=prec_ac)
+        for line in parsed_inv.get("lines", []):
+            line["qty"] = float_round(line["qty"], precision_digits=prec_uom)
```
There is an exception possibility here:

Invoice line qty = 0,5 / price = 100
Product Unit of Measure = 0

When importing the quantity will be set to 1, but the price is not changed accordingly. Result is an unclear debit/credit error from Odoo.

StefanRijnhart replied:
I think this will now lead to an adjustment line here: https://github.com/OCA/edi/blob/60742e0cf64807a4981cdd13fe2e51f51235b4a8/account_invoice_import/components/edi_input_process_business_document_account_invoice.py#L291. Is that alright for you?

vrenaville committed:
[RM] invoice line doesn't exist on V13

CasVissers-360ERP commented on account_invoice_import/wizard/account_invoice_import.py:
```diff
@@ -0,0 +1,1168 @@
+            parsed_inv.get("currency"), parsed_inv["chatter_msg"]
+        )
+        journal_id = (
+            aio.with_context(type=parsed_inv["type"], company_id=company.id)
```
Should be default_type

StefanRijnhart replied:
I think this comment still applies here: https://github.com/OCA/edi/blob/60742e0cf64807a4981cdd13fe2e51f51235b4a8/account_invoice_import/components/edi_input_process_business_document_account_invoice.py#L71

CasVissers-360ERP reviewed (changes requested):
Additional feedback

StefanRijnhart commented:
@CasVissers-360ERP I'm impressed but also confused that you review both PRs for the same module and the same Odoo version ;-) Should we try to converge the two efforts?

On the other PR, @etobella has proposed a refactoring to integrate with the `edi` module. How does everybody here feel about that?

yvaucher commented:
FTR I currently have issues with this PR with tax included.

yvaucher commented:
@StefanRijnhart if suited to integrate it more with the base module I'm in favor of it.

@etobella I'm very interested in your proposed changes.

This current PR is an attempt to migrate it and AFAIU @NL66278 tries to solve some issues in it. (Don't know which ones thought.)

I'll update this PR according to the current reviews.

yvaucher commented:
thanks @astirpe for the kind review, I just updated the PR solving the 2 issues you raised.

yvaucher commented:
Seems https://github.com/NL66278/edi/pull/1 is the most advanced version, I'll give it a try.

NL66278 commented:
I did some refactoring to silence the 'too complicated' lint check and to make the code easier to maintain. Also solving some other problems the linter was complaining about, like shadowing the import module config with a variable of the same name in a function.

yvaucher commented:
@NL66278 yep I saw it, I just cherry picked your 3 commits 662651e, 3d7afe3 and 8b0560e

I splitted and squashed the others about black to reduce the number of commits.

NL66278 commented:
Code must be blacked again...

yvaucher commented:
@NL66278 seems there was some ping pong between my black commit and yours.

I removed the formatting from: https://github.com/OCA/edi/pull/213/commits/bb023af52362e9f4c862f1b7b86fab3e7ff45b37

As I'm running `pre-commit run -a` I think this last version should be fine.

NL66278 commented:
I think it now failes on this line:
base_business_document_import/models/business_document_import.py:83:5: C901 'BusinessDocumentImport._match_partner' is too complex (25)

For the rest an enormous amount of prettier messages. Annoying, but as prettier passes I think not the reason for failing Travis.

yvaucher committed:
[REF] base_business_document_import: matching

Reduce _match_partner complexity

yvaucher commented:
@NL66278 last commit aims to solve the complexity of `_match_partner`

NL66278 commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -48,7 +48,73 @@ def _strip_cleanup_dict(self, match_dict):
+                    ("country_id", "=", False),
+                    ("country_id", "=", country.id),
+                ]
+            else:
```
No need for an else here, as you do an early return in the if branch.

NL66278 commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -48,7 +48,73 @@ def _strip_cleanup_dict(self, match_dict):
+    def _get_match_partner_order(self, partner_type):
+        if partner_type == "supplier":
+            return "supplier_rank desc"
+        elif partner_type == "customer":
```
Can just be if

NL66278 commented on base_business_document_import/models/business_document_import.py:
```diff
@@ -48,7 +48,73 @@ def _strip_cleanup_dict(self, match_dict):
+    def _get_match_partner_type_label(self, partner_type):
+        if partner_type == "supplier":
+            return _("supplier")
+        elif partner_type == "customer":
```
Can just be if

CasVissers-360ERP commented:
Some additional functional feedback: I think it's would be good to run _onchange_partner_id after creating the invoice. I.e. partner_bank_id is not set.

yvaucher committed:
[IMP] partner_identification_import: align _match_partner_contact signature

yvaucher committed:
Remove non serializable data

yvaucher commented:
Fixed match by partner email, I had messed up the refactor on this. The logic is now restored on the search using the domain name of the email.

yvaucher committed:
Add a unit test with tax not included

yvaucher committed:
Add pytest compatibility

i-vyshnevska committed:
[IMP] Make settings configurable

i-vyshnevska committed:
[IMP] improvements

i-vyshnevska committed:
[IMP] Make tax control configurable

Some countries don't provide tax data in invoices, but invoices
by themselves can have the price included tax. In this case, there will be
attempt to create an adjustment line which leads to increasing total
invoice amount and complete fail.

i-vyshnevska committed:
[IMP] Add sequence to not fill names manually

alexis-via commented:
Arf, commit bb023af52362e9f4c862f1b7b86fab3e7ff45b37 just removed the MOST IMPORTANT PART OF THE CODE: the documentation of the pivot format !!! If you remove that, then you can't understand how this module works!

yvaucher commented:
@alexis-via I amended the commit https://github.com/OCA/edi/pull/213/commits/0a5765c600a99b64657df0c7d985c95ceb470931 to restore the commented section for usage.

bosd commented:
Looks good afaik, could not do full functional test without supporting modules.
noticed that it also commits files of the module  ```base_business_document``` 
Is this version with the merging of edi the way forward?

bosd commented:
Please include https://github.com/OCA/edi/pull/84/

yvaucher commented:
@bosd the edi part is separated in further versions. We might want to not make the v13 an exception, thus I have to rework this to extract EDI.

For #84 I don't know how to integrate it as I don't see unit tests and don't know what to test. I suggest that we publish a first version without it first.

NL66278 reviewed (approved):
:+1: LGTM

bosd commented:
@astirpe your change request is still showing a red icon next to your name. Is it oke now?

yankinmax committed:
[FIX] account_invoice_import: fix external_id for Vendor bills action

mylbco commented:
Can this PR be merged ?

yvaucher committed:
Merge pull request #4 from yankinmax/13.0-mig-account_invoice_import

[FIX] account_invoice_import: fix external_id for Vendor bills action

StefanRijnhart commented:
@yvaucher please install the pre-commit hook in your local repo, and commit its changes.

yvaucher commented:
@StefanRijnhart I most likely did, those might come from the last added commit, will have to fix that.

olgamarcocb commented on account_invoice_import/components/edi_input_process_business_document_account_invoice.py:
```diff
@@ -0,0 +1,551 @@
+class EDIInputProcessBusinessDocumentImport(Component):
+    _name = "edi.input.process.account.invoice.import"
+    _inherit = "edi.input.process.business.document.import"
+    _usage = "edi.input.process.account_invoice_import"
```
_usage = "input.process"
    _backend_type = "account_invoice_import"

olgamarcocb commented on account_invoice_import/components/edi_input_process_business_document_account_invoice.py:
```diff
@@ -0,0 +1,551 @@
+            parsed_inv.get("currency"), parsed_inv["chatter_msg"]
+        )
+        journal_id = (
+            amo.with_context(type=parsed_inv["type"], company_id=company.id)
```
with_context(default_type=...

olgamarcocb commented on account_invoice_import/components/edi_input_process_business_document_account_invoice.py:
```diff
@@ -0,0 +1,551 @@
+    def _prepare_update_invoice_vals(self, parsed_inv, invoice):
+        bdio = self.env["business.document.import"]
+        vals = {
+            "ref": parsed_inv.get("invoice_number"),
```
invoice_payment_ref

olgamarcocb reviewed (commented):
Please check!

Martronic-SA commented:
Hi!
I'm sorry but I tried to install this module on a Odoo CE v13 and I've had a lot of errors (missing edi dependencies for components...)
Finally I've tried to backport the v14.0 module and, except 2 -3 small issues (security with a non existing group, 2 views and "move_type" instead of "type"), I've managed to install it and it seems to be working as expected.

simahawk commented:
what's the status here? @yvaucher could you check pre-commit pls?

bosd commented:
ping @yvaucher Can you check the pre-commit?

bosd commented:
Can a maintainer fix pre-commit on this one?

yvaucher commented:
rebased

yvaucher commented:
Can't reopen this one, opening #727
